src/plugins/intel_gpu/src/graph/impls/onednn/gemm_onednn.cpp

```diff
@@ -186,8 +186,15 @@ struct gemm_onednn : typed_primitive_onednn_impl<gemm> {
             if (ret) {
                 tag = convert_data_format(transposed_format);
                 dnnl::memory::dims original_dims = dims;
-                for (size_t i = 0; i < original_dims.size(); ++i) {
-                    dims[i] = original_dims[order[i]];
+                if (is_input) {
+                    for (size_t i = 0; i < original_dims.size(); ++i) {
+                        dims[i] = original_dims[order[i]];
+                    }
+                } else {
+                    // Get non-transposed dims for output dims
+                    for (size_t i = 0; i < original_dims.size(); ++i) {
+                        dims[order[i]] = original_dims[i];
```

Comment:
Didn't you say that output only changes the tag and uses the dims order as is? Is it necessary? If true, it needs different behavior vs input. right?

Reply:
Before the get_gemm_primitive_md(), the output shape is already transposed by transform process. And the get_gemm_primitive_md() tries to transpose again with the already transposed output shape. This code is to get non-transposed output shape from transposed shape which is already done by transform.